pyvista/core/utilities/features.py

```diff
@@ -99,6 +99,121 @@ def voxelize(mesh, density=None, check_surface=True):
     return vox
 
 
+def voxelize_volume(mesh, density=None, check_surface=True):
+    """Voxelize mesh to create a RectilinearGrid voxel volume.
+
+    Creates a voxel volume that encloses the input mesh and discretizes the cells
+    within the volume that intersect and are contained within the input mesh.
+    All discretized cells are named ``MeshCells``. All other cells are named ``BackgroundCells``.
+
+    Parameters
+    ----------
+    mesh : pyvista.DataSet
+        Mesh to voxelize.
+
+    density : float | array_like[float]
+        The uniform size of the voxels when single float passed.
+        Nonuniform voxel size if a list of values are passed along x,y,z directions.
+        Defaults to 1/100th of the mesh length.
+
+    check_surface : bool, default: True
+        Specify whether to check the surface for closure. If on, then the
+        algorithm first checks to see if the surface is closed and
+        manifold. If the surface is not closed and manifold, a runtime
+        error is raised.
+
+    Returns
+    -------
+    pyvista.RectilinearGrid
+        RectilinearGrid as voxelized volume with discretized cells.
+
+    Notes
+    -----
+    Prior to version 0.39.0, this method improperly handled the order of
+    structured coordinates.
+
+    Examples
+    --------
+    Create an equal density voxel volume from input mesh.
+
+    >>> import pyvista as pv
+    >>> import numpy as np
+
+    Load file from PyVista examples.
+
+    >>> from pyvista import examples
+    >>> mesh = examples.download_cow()
+
+    Create an equal density voxel volume and plot the result.
+
+    >>> vox = pv.voxelize_volume(mesh, density=0.15)
+    >>> cpos = [(15, 3, 15), (0, 0, 0), (0, 0, 0)]
+    >>> vox.plot(scalars='MeshCells', show_edges=True, cpos=cpos)
+
+    Slice the voxel volume to view ``MeshCells``.
+
+    >>> slices = vox.slice_orthogonal()
+    >>> slices.plot(scalars='MeshCells', show_edges=True)
+
+    Create a voxel volume from unequal density dimensions and plot result.
+
+    >>> vox = pv.voxelize_volume(mesh, density=[0.15, 0.15, 0.5])
+    >>> vox.plot(scalars='MeshCells', show_edges=True, cpos=cpos)
+
+    Slice the unequal density voxel volume to view ``MeshCells``.
+
+    >>> slices = vox.slice_orthogonal()
+    >>> slices.plot(scalars='MeshCells', show_edges=True, cpos=cpos)
+
+    """
+    if not pyvista.is_pyvista_dataset(mesh):
+        mesh = wrap(mesh)
```

Comment:
```suggestion
    if not pyvista.is_pyvista_dataset(mesh):
        mesh = wrap(mesh)
```
We don't need to check `is_pyvista_dataset` as it is checked in `wrap` function.
https://github.com/pyvista/pyvista/blob/35e0063ccbc8053a8edf9d8ac6d5fada339c2a90/pyvista/core/utilities/helpers.py#L102-L104